### PR TITLE
updated expo/metro-config

### PR DESCRIPTION
**Pull request by EvanBacon**
# Why

- This has updates to expo/config (related to the dropping of .expo support) and updates to debugging.

### CODE REVIEW

**Comment by EvanBacon** [comment]
I'm getting weird node_modules errors when trying to use this in NCL.
```
Error: Cannot find module 'jsonfile/utils'
Require stack:
- /Users/evanbacon/Documents/GitHub/expo/node_modules/@expo/metro-config/node_modules/@expo/configure-splash-screen/node_modules/fs-extra/lib/json/output-json.js
- /Users/evanbacon/Documents/GitHub/expo/node_modules/@expo/metro-config/node_modules/@expo/configure-splash-screen/node_modules/fs-extra/lib/json/index.js
- /Users/evanbacon/Documents/GitHub/expo/node_modules/@expo/metro-config/node_modules/@expo/configure-splash-screen/node_modules/fs-extra/lib/index.js
- /Users/evanbacon/Documents/GitHub/expo/node_modules/@expo/metro-config/node_modules/@expo/configure-splash-screen/build/validators/utils.js
- /Users/evanbacon/Documents/GitHub/expo/node_modules/@expo/metro-config/node_modules/@expo/configure-splash-screen/build/validators/index.js
- /Users/evanbacon/Documents/GitHub/expo/node_modules/@expo/metro-config/node_modules/@expo/configure-splash-screen/build/ios/index.js
- /Users/evanbacon/Documents/GitHub/expo/node_modules/@expo/metro-config/node_modules/@expo/configure-splash-screen/build/index.js
- /Users/evanbacon/Documents/GitHub/expo/node_modules/@expo/metro-config/node_modules/@expo/config-plugins/build/android/SplashScreen.js
- /Users/evanbacon/Documents/GitHub/expo/node_modules/@expo/metro-config/node_modules/@expo/config-plugins/build/android/index.js
- /Users/evanbacon/Documents/GitHub/expo/node_modules/@expo/metro-config/node_modules/@expo/config-plugins/build/index.js
- /Users/evanbacon/Documents/GitHub/expo/node_modules/@expo/metro-config/node_modules/@expo/config/build/plugins/withConfigPlugins.js
- /Users/evanbacon/Documents/GitHub/expo/node_modules/@expo/metro-config/node_modules/@expo/config/build/Config.js
- /Users/evanbacon/Documents/GitHub/expo/node_modules/@expo/metro-config/node_modules/@expo/config/build/index.js
- /Users/evanbacon/Documents/GitHub/expo/node_modules/@expo/metro-config/build/ExpoMetroConfig.js
- /Users/evanbacon/Documents/GitHub/expo/packages/expo-updates/scripts/createManifest.js
    at Function.Module._resolveFilename (internal/modules/cjs/loader.js:880:15)
    at Function.Module._load (internal/modules/cjs/loader.js:725:27)
    at Module.require (internal/modules/cjs/loader.js:952:19)
    at require (internal/modules/cjs/helpers.js:88:18)
    at Object.<anonymous> (/Users/evanbacon/Documents/GitHub/expo/node_modules/@expo/metro-config/node_modules/@expo/configure-splash-screen/node_modules/fs-extra/lib/json/output-json.js:3:23)
    at Module._compile (internal/modules/cjs/loader.js:1063:30)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:1092:10)
    at Module.load (internal/modules/cjs/loader.js:928:32)
    at Function.Module._load (internal/modules/cjs/loader.js:769:14)
    at Module.require (internal/modules/cjs/loader.js:952:19) {
  code: 'MODULE_NOT_FOUND',
  requireStack: [
    '/Users/evanbacon/Documents/GitHub/expo/node_modules/@expo/metro-config/node_modules/@expo/configure-splash-screen/node_modules/fs-extra/lib/json/output-json.js',
    '/Users/evanbacon/Documents/GitHub/expo/node_modules/@expo/metro-config/node_modules/@expo/configure-splash-screen/node_modules/fs-extra/lib/json/index.js',
    '/Users/evanbacon/Documents/GitHub/expo/node_modules/@expo/metro-config/node_modules/@expo/configure-splash-screen/node_modules/fs-extra/lib/index.js',
    '/Users/evanbacon/Documents/GitHub/expo/node_modules/@expo/metro-config/node_modules/@expo/configure-splash-screen/build/validators/utils.js',
    '/Users/evanbacon/Documents/GitHub/expo/node_modules/@expo/metro-config/node_modules/@expo/configure-splash-screen/build/validators/index.js',
    '/Users/evanbacon/Documents/GitHub/expo/node_modules/@expo/metro-config/node_modules/@expo/configure-splash-screen/build/ios/index.js',
    '/Users/evanbacon/Documents/GitHub/expo/node_modules/@expo/metro-config/node_modules/@expo/configure-splash-screen/build/index.js',
    '/Users/evanbacon/Documents/GitHub/expo/node_modules/@expo/metro-config/node_modules/@expo/config-plugins/build/android/SplashScreen.js',
    '/Users/evanbacon/Documents/GitHub/expo/node_modules/@expo/metro-config/node_modules/@expo/config-plugins/build/android/index.js',
    '/Users/evanbacon/Documents/GitHub/expo/node_modules/@expo/metro-config/node_modules/@expo/config-plugins/build/index.js',
    '/Users/evanbacon/Documents/GitHub/expo/node_modules/@expo/metro-config/node_modules/@expo/config/build/plugins/withConfigPlugins.js',
    '/Users/evanbacon/Documents/GitHub/expo/node_modules/@expo/metro-config/node_modules/@expo/config/build/Config.js',
    '/Users/evanbacon/Documents/GitHub/expo/node_modules/@expo/metro-config/node_modules/@expo/config/build/index.js',
    '/Users/evanbacon/Documents/GitHub/expo/node_modules/@expo/metro-config/build/ExpoMetroConfig.js',
    '/Users/evanbacon/Documents/GitHub/expo/packages/expo-updates/scripts/createManifest.js'
  ]
}
```

File: packages/expo-updates/CHANGELOG.md [hunk 1]
```diff
@@ -6,6 +6,7 @@
 
 ### 🎉 New features
 
+- Updated `@expo/metro-config`. ([#12252](https://github.com/expo/expo/pull/12252) by [@EvanBacon](https://github.com/EvanBacon))
```
Comment (ide) [inline packages/expo-updates/CHANGELOG.md:9]: would be helpful to describe the version upgrade and anything meaningful that's new

**Comment by shamilovtim** [comment]
🤔
code: 'MODULE_NOT_FOUND',
Error: Cannot find module 'core-js/es/string/match-all'

I wiped out node_modules, deleted yarn.lock and downgraded to reanimated 1.X. Seems to have fixed it. No fancy animations in the particular app I'm working on, so not a huge a deal for now.

**Comment by brentvatne** [comment]
> I wiped out node_modules, deleted yarn.lock

i'd be curious if you tried that without downgrading if that's resolve your issue

**Comment by ide** [comment]
> is it a yarn workspace? sometimes dependency hoisting .. goes wrong

To add some more color, I've found that with Yarn workspaces it helps to get all of the workspace packages using the same versions of the same dependencies where possible. `resolutions` is another wrench. Otherwise a lot of strange errors can occur, either when installing the dependencies or running a bundler. Maybe Yarn 2 fixes this because/but it's a significantly different piece of software.

**Comment by shamilovtim** [comment]
> > I wiped out node_modules, deleted yarn.lock
> 
> i'd be curious if you tried that without downgrading if that's resolve your issue. is it a yarn workspace? sometimes dependency hoisting .. goes wrong

nope just a plan old yarn project, pretty much as vanilla as they come.

**Pull request merged**